build-tools/packages/build-cli/src/codeCoverageAnalysisTool/getCoverageMetrics.ts

```diff
@@ -0,0 +1,148 @@
+	const report: CoverageReport[] = [];
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const coverageForPackagesResult = result?.coverage?.packages?.[0]?.package as any[];
+
+	for (const coverageForPackage of coverageForPackagesResult) {
+		const packageName = coverageForPackage["$"].name as string;
+		const lineCoverage = Number.parseFloat(coverageForPackage["$"]["line-rate"]);
+		const branchCoverage = Number.parseFloat(coverageForPackage["$"]["branch-rate"]);
+		report.push({
+			packageName,
+			lineCoverage,
+			branchCoverage,
+		});
+	}
+
+	return report;
+};
+
+/**
+ * Method that returns the coverage report for the baseline build
+ * @param baselineZip - zipped coverage files for the baseline build
+ * @returns an array of coverage metrics for baseline containing packageName, lineCoverage and branchCoverage
+ */
+export const getCoverageMetricsForBaseline = async (
+	baselineZip: JSZip,
+): Promise<CoverageReport[]> => {
+	const coverageReportsFiles: string[] = [];
+	baselineZip.forEach((filePath) => {
+		if (filePath.endsWith("cobertura-coverage-patched.xml"))
+			coverageReportsFiles.push(filePath);
+	});
+
+	const coverageMetricsForBaseline: CoverageReport[] = [];
+	const xmlParser = new Parser();
+
+	try {
+		console.log(`${coverageReportsFiles.length} coverage files found in baseline`);
+
+		for (const coverageReportFile of coverageReportsFiles) {
+			const jsZipObject = baselineZip.file(coverageReportFile);
+			if (!jsZipObject) {
+				console.log(`could not find file ${coverageReportFile}`);
+			}
+
+			const coverageReportXML = await jsZipObject?.async("nodebuffer");
+			if (coverageReportXML !== undefined) {
+				// eslint-disable-next-line @typescript-eslint/no-explicit-any -- missing type for XML output
+				xmlParser.parseString(coverageReportXML, (err: Error | null, result: any): void => {
+					if (err) {
+						console.warn(`Error processing file ${coverageReportFile}: ${err} in baseline`);
+						return;
+					}
+					const metrics = extractCoverageMetrics(result);
+					metrics.forEach((metric: CoverageReport) => {
+						if (
+							metric.packageName &&
+							!Number.isNaN(metric.lineCoverage) &&
+							!Number.isNaN(metric.branchCoverage) &&
+							metric.lineCoverage < 1
+						) {
+							coverageMetricsForBaseline.push(metric);
+						}
+					});
+				});
+			}
+			if (coverageMetricsForBaseline.length > 0) {
+				break;
+			}
+		}
+	} catch (error) {
+		console.log(`Error encountered with reading files: ${error} in baseline`);
+	}
+
+	console.log(`${coverageMetricsForBaseline.length} coverage reports generated`);
+	return coverageMetricsForBaseline;
+};
+
+/**
+ * Method that returns the coverage metrics for the pr
+ * @param coverageReportsFolder - The folder where the coverage reports for the pr can be found
+ * @returns an array of coverage metrics for baseline containing packageName, lineCoverage and branchCoverage
+ */
+export const getCoverageMetricsForPr = async (
```

Comment:
The function is called getCoverageMetrics, but the type it returns is CoverageReport - suggest unifying the names.